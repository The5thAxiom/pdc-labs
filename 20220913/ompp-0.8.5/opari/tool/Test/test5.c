#include <stdio.h>

int main() {
  int i, j, k, l;
  #pragma omp parallel         /* parallel */ \
      for                      /* do */ \
      lastprivate(k)           /* do */ \
      private(i,j)             /* parallel */ \
      lastprivate              /* do */ \
      (                        /* do */ \
        l                      /* do */ \
      ) schedule(dynamic       /* do */ \
      )
  for(i=0; i<4;++i) {
    printf("pdo %d\n", i);
  }
}
