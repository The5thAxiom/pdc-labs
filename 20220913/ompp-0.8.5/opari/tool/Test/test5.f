      program test
      integer i,j,k,l

!$omp  parallel
!$omp+ sections
!$omp+ default(shared) private(i) lastprivate(k)
!$omp+ lastprivate(l)
!$omp  section
         write(*,*) "section1"
!$omp  section
         write(*,*) "section2"
!$omp  end parallel sections

      end
