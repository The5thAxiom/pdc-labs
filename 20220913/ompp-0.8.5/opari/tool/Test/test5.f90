      program test5
      integer i, j, k, l
      !$omp parallel                          & !parallel
      !$omp & do                              & !do
      !$omp & lastprivate(k)                  & !do
      !$omp & private(i,j)                    & !parallel
      !$omp & lastprivate                     & !do
      !$omp & (                               & !do
      !$omp &   l                             & !do
      !$omp & ) schedule(dynamic              & !do
      !$omp   )
      do i=1,4
        write(*,*) "pdo", i
      enddo
      !$omp end parallel do
      end
