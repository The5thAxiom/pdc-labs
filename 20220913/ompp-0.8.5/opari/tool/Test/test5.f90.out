#ifdef _POMP
#  undef _POMP
#endif
#define _POMP 200110

#line 1 "test5.f90"
      program test5
      integer i, j, k, l
      call POMP_Parallel_fork(23)
#line 3 "test5.f90"
      !$omp parallel                          & !parallel
      !$omp & private(i,j)                      !parallel
      call POMP_Parallel_begin(23)
      call POMP_Do_enter(23)
#line 3 "test5.f90"
      !$omp   do                              &    
      !$omp & lastprivate(k)                  &    
      !$omp & lastprivate                     &    
      !$omp & (                               &    
      !$omp &   l                             &    
      !$omp & ) schedule(dynamic              &    
      !$omp   )
      do i=1,4
        write(*,*) "pdo", i
      enddo
!$omp end do nowait
      call POMP_Barrier_enter(23)
!$omp barrier
      call POMP_Barrier_exit(23)
      call POMP_Do_exit(23)
      call POMP_Parallel_end(23)
!$omp end parallel
      call POMP_Parallel_join(23)
#line 16 "test5.f90"
      end
